### DISCUSSION TITLE
How do I configure WSGI parameters in Ckan 2.8?

### DISCUSSION THREAD

**Discussion by pauloneves** [ckan/ckan#6185] [Q&A]
I'm trying to publish a big CSV file in my CKAN 2.8 installation. It publishes fine and I can see the preview in the Data Explorer. 

It fails when I try to download it in a converted format from the menu of the "Download" button in formats like Json, TSV or XML.

Every time I try to download, I get the following error:

```
504 Gateway Time-out
The server didn't respond in time. 
```

After a lot of research, I believe that it will be fixed if I configure the `harakiri` WSGI config option. 

How can I configure the WSGI options in CKAN 2.8?

**Comment by pauloneves** [comment]
I timed it and the 504 error appears after exactly 30 seconds. I believe this is some kind of  timeout configuration. Any idea how to increase this timeout in CKAN 2.8?

**Reply by pauloneves** [reply]
I'm also posting it to Stack Overflow: https://stackoverflow.com/questions/68106025/ckan-how-to-fix-error-504-when-serving-a-big-file

**Comment by anuveyatsu** [comment]
@pauloneves I haven't experienced such an issue but could you try to change your uwsgi config options? There should be parameters for setting up timeouts I guess.

**Comment by pauloneves** [comment]
Sure, but I don't know to configure it in CKAN 2.8. Ckan 2.9 has the
harakiri option ,but I don't know to do it in this CKAN version

On Mon, Jun 28, 2021 at 2:59 AM Anuar Ustayev ***@***.***>
wrote:

> @pauloneves <https://github.com/pauloneves> I haven't experienced such an
> issue but could you try to change your uwsgi config options? There should
> be parameters for setting up timeouts I guess.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ckan/ckan/discussions/6185#discussioncomment-931086>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABV7HB7DKEJA6YFZX3NMEDTVAFTBANCNFSM47ETXM5Q>
> .
>


-- 
Paulo Eduardo Neves

**Reply by anuveyatsu** [reply]
@pauloneves Please take a look at stackoverflow where you have couple of suggestions 😄

**Comment by jqnatividad** [comment]
In your [Apache config file](https://docs.ckan.org/en/2.8/maintaining/installing/deployment.html#create-the-apache-config-file), have you tried adding the [`request-timeout`](https://modwsgi.readthedocs.io/en/develop/configuration-directives/WSGIDaemonProcess.html#) option to `WSGIDaemonProcess`?

From:
`WSGIDaemonProcess ckan_default display-name=ckan_default processes=2 threads=15`
to
`WSGIDaemonProcess ckan_default display-name=ckan_default processes=2 threads=15 request-timeout=60`
